### ISSUE TITLE
Abstract

### ISSUE THREAD

**Issue by colah** [thomas-tanay/post--L2-regularization#10]
# General Comments

At the moment, your article opens with a short section giving a flavor of the content of the article:

![image](https://user-images.githubusercontent.com/61658/31201969-834b0444-a915-11e7-9b54-18fb475221d8.png)

I think this is a really good direction, but is trying to be too many things at once right now. In particular, it feels like it's playing a balancing act between being a visual abstract (which I think would try to be more focused and brief) and an introduction (which would provide more context).

In doing so, it falls into a bit of an uncanny valley: It doesn't quite have enough context to really get what's going on if you come in knowing about adversarial examples but not the ideas in this article. Readers will suspend the expectation of really knowing what's going on for a cool "teaser" at the top of an article, but once it starts to feel like a full section they get confused.

My inclination is to tighten this into a visual abstract / teaser / hero, and use the background section as an introduction (which it already is and does quite well).

# Specific advice:

**1. Figure out the angle / core point**

There's a lot of interesting things about your article:

* A geometric way of thinking about adversarial examples
* Adversarial examples can occur in two dimensions.
* Conflict between minimizing training error and reducing adversarial distance.
* L2 regularization can help control this balance.
* Finding this helps for both linear models and simple neural networks.
* ...

I think you want to pick out a couple that are interesting, hook the reader, and give the flavor of your article. 

**2. Focus**

Then you want to boil it down into a few sentences and an interactive. At the very dense end, you could do something like this:

![image](https://user-images.githubusercontent.com/61658/31202906-80737f22-a919-11e7-90d4-157ff2d8ba93.png)

But I'd avoid trying to squeeze in that much text.

**3. Include a bit more context if possible.**

This is in tension with other things I've said, but I think you could give a bit more context. That said, I think you'll also make it a lot less important by making it shorter and queuing the reader that they don't have to understand everything that's going on here.

**4. Make it interactive, not a video.**

The interactive at the top is kind of a video right now. You hit play and it goes. If possible, it would be better for the reader to be able to fiddle directly with the regularization parameter. 

# Examples

Here's one example of what a tight visual abstract might look like. It cuts a lot of nice stuff, but you have lots of space to talk about the content you didn't mention here later in the article. The goal at this point is to interest the reader.

![image](https://user-images.githubusercontent.com/61658/31203940-3679e1c2-a91e-11e7-94a5-20f9b3af91e4.png)

Or with a different focus:

![image](https://user-images.githubusercontent.com/61658/31204232-c5d0b6ba-a91f-11e7-8ee6-c408b520d341.png)

**Comment by thomas-tanay** [comment]
These are all very good points. My initial objective was indeed to make a short visual abstract, but I got caught into trying to summarize everything. 

I will probably go for something like your first suggestion:

![31203940-3679e1c2-a91e-11e7-94a5-20f9b3af91e4](https://user-images.githubusercontent.com/23214134/31237488-02c952f0-a9ef-11e7-970f-4920614f9e77.png)

(showing the tilting plane animation on MNIST is actually what motivated me to write this post in the first place -- I consider it to be the core idea of the article)

**Comment by thomas-tanay** [comment]
I just submitted a new version with a simplified abstract:

![abstract](https://user-images.githubusercontent.com/23214134/31493216-1a2cb21e-af46-11e7-9c51-92f4844d1c06.png)

I am not completely satisfied with the design though, I will try to improve it in the coming days.

**Comment by thomas-tanay** [comment]
Here is what I have now:

![abstract](https://user-images.githubusercontent.com/23214134/31508474-18dae054-af76-11e7-9b3c-e3a46dd3080c.png)

**Comment by colah** [comment]
That looks great!

I think it might be possible to improve the alignment of the text and the diagram, but I'm not sure exactly how to do it. Shan may have thoughts.

It would add a little noise, but you could consider trying to visually display adversarial distance some how.

**Comment by colah** [comment]
A new version while I comment!

Does the orange represent adversarial distance? Or a margin between data points? (It feels a bit too thin to be the average adversarial distance.)

I really like including the adversarial distance or something closely related to it somehow. This seems like one good candidate. Another possibility might be to draw a single adversarial example...

**Comment by thomas-tanay** [comment]
ah that's a good point. At the moment the orange is the margin but I don't refer to it that much in the text.

I could replace that by the adversarial distance, which is the distance between each mean digit and the boundary.

**Comment by colah** [comment]
Yeah, I feel like adversarial distance is probably a more important idea for you than the margin...

The orange may git a bit overwhelming for adversarial distance though. You might need to experiment a bit.

**Comment by thomas-tanay** [comment]
I updated it so that the shaded area is now the adversarial distance. I'm still not completely sure about the colors, I might change them at some point.

![abstract](https://user-images.githubusercontent.com/23214134/31511465-ba4aab28-af7f-11e7-9452-eec18f7df222.png)

**Comment by colah** [comment]
Looking nice!

* Is it true that adversarial distance is the distance to the means? Not obvious that "average distance" = "distance to averages", if it's possible for one point to be on the other side of the line...

* Right now, the blue boxes on each side have a lot of visual weight. I'm not sure what the right thing to do is, but I'd guess it should be something different.

**Comment by thomas-tanay** [comment]
- It's true that the relation: d_adv = "distance to averages" is not obvious, but it works because d_adv is an average of signed distances. I show that the equation is true in the subsection "Adversarial distance and tilting angle": d_adv = 1/2 || j - i || cos(theta)

- I agree that the blue boxes are too salient. Maybe very light grays work better?

![abstract](https://user-images.githubusercontent.com/23214134/31520701-a39220cc-af9e-11e7-81ab-9846793eb072.png)
